dgl_sparse/src/sparse_matrix.cc

```diff
@@ -122,6 +124,57 @@ c10::intrusive_ptr<SparseMatrix> SparseMatrix::FromDiag(
   return SparseMatrix::FromDiagPointer(diag, value, shape);
 }
 
+c10::intrusive_ptr<SparseMatrix> SparseMatrix::IndexSelect(
+    int64_t dim, torch::Tensor ids) {
+  auto id_array = TorchTensorToDGLArray(ids);
+  bool rowwise = dim == 0;
+  SparseFormat sparse_format;
```

Comment:
The only usage of sparse_format is L144, why not check rowwise in L144 directly?

Reply:
Ok, we have changed this many times, because we want to make it more readable at the very beginning.